jdaviz/app.py

```diff
@@ -545,15 +545,69 @@ def get_subsets_from_viewer(self, viewer_reference, data_label=None, subset_type
                                          cls=None)
         regions = {}
 
+        def _get_all_subregions(mask, spec_axis_data):
+            """
+            Return all subregions within a subset.
+
+            Parameters
+            ----------
+            mask : list
+                List of indices in spec_axis_data that are part of the subset.
+            spec_axis_data : list
+                List of spectral axis values.
+            Returns
+            -------
+            combined_spec_region : `~specutils.SpectralRegion`
+                SpectralRegion object containing all subregions of the subset.
+            """
+            if len(mask) == 0:
+                # Mask should only be 0 if ApplyROI is used to incorrectly
+                # create subsets via the API
+                raise ValueError("Mask has length 0, ApplyROI may have been used incorrectly")
+
+            current_edge = 0
+            combined_spec_region = None
+            for index in range(1, len(mask)):
+                # Find spot where mask == True is for a different region of the subset
+                # i.e. mask = [0, 1, 4, 5]
+                # mask[2] != mask[1] + 1
+                if mask[index] != mask[index - 1] + 1:
```

Comment:
This assumes that `mask` is a well-behaved list (no repeats, monotonically increasing)? What if someone inputs crazy values like `mask = [-1, 14, -10, 0]`? Is this a use case we need to worry about?

Reply:
The logic there is ok, because `mask` there is actually the indices where the mask derived from the subset on the spectrum is `True` - the variable name is a little confusing. The code fed into that `mask` argument is `np.where(value.to_mask() == True)`, which should work fine based on the `Subset.to_mask()` logic, I think.